### PR TITLE
Expose types with Parameters for DESCRIBE INPUT

### PR DESCRIPTION
**Pull request by Praveen2112**

### CODE REVIEW

File: presto-tests/src/test/java/io/prestosql/tests/AbstractTestEngineOnlyQueries.java [hunk 1]
```diff
@@ -730,7 +730,7 @@ public void testDescribeInput()
         MaterializedResult expected = resultBuilder(session, BIGINT, VARCHAR)
                 .row(0, "unknown")
                 .row(1, "bigint")
-                .row(2, "varchar")
+                .row(2, "varchar(25)")
```
Comment (kokosing) [inline presto-tests/src/test/java/io/prestosql/tests/AbstractTestEngineOnlyQueries.java:733]: It looks like we are missing dedicated test for this feature. Can you please add it where you could also test other parametric types including timestamp, char and decimal?

Comment (Praveen2112) [reply presto-tests/src/test/java/io/prestosql/tests/AbstractTestEngineOnlyQueries.java:733]: Thanks for your insight. Fixed it

File: presto-tests/src/test/java/io/prestosql/tests/AbstractTestEngineOnlyQueries.java [hunk 2]
```diff
@@ -730,7 +730,18 @@ public void testDescribeInput()
...
+                .build();
+        assertEqualsIgnoreOrder(actual, expected);
+
+        session = Session.builder(getSession())
+                .addPreparedStatement("my_query", "SELECT 1 from (VALUES (CHAR 'Pi', TIMESTAMP '2012-03-14 1:59:26.535', DECIMAL '3.14')) AS t (t_varchar, t_timestamp, t_decimal) WHERE t_varchar = ? AND t_timestamp = ? AND t_decimal = ?")
+                .build();
+        actual = computeActual(session, "DESCRIBE INPUT my_query");
+        expected = resultBuilder(session, BIGINT, VARCHAR)
+                .row(0, "char(2)")
+                .row(1, "timestamp(3)")
```
Comment (kokosing) [inline presto-tests/src/test/java/io/prestosql/tests/AbstractTestEngineOnlyQueries.java:743]: can you have a test with  timestamp(6) as well as unbound varchar?

Comment (Praveen2112) [reply presto-tests/src/test/java/io/prestosql/tests/AbstractTestEngineOnlyQueries.java]: I have added test for timestamp and unbounded varchar

File: presto-tests/src/test/java/io/prestosql/tests/AbstractTestEngineOnlyQueries.java [hunk 3]
```diff
@@ -730,7 +730,24 @@ public void testDescribeInput()
...
-                .row(2, "varchar")
+                .row(2, "varchar(25)")
+                .build();
+        assertEqualsIgnoreOrder(actual, expected);
+
+        session = Session.builder(getSession())
+                .addPreparedStatement(
+                        "my_query",
+                        "SELECT 1 from " +
+                        "    (VALUES (CHAR 'Pi', CAST ('PI' AS VARCHAR), TIMESTAMP '2012-03-14 1:59:26.535', TIMESTAMP '2012-03-14 1:59:26.535897', DECIMAL '3.14')) AS t (t_char, t_varchar, t_timestamp, t_timestamp_2, t_decimal) " +
```
Comment (electrum) [inline presto-tests/src/test/java/io/prestosql/tests/AbstractTestEngineOnlyQueries.java:741]: This is hard to read due to the length -- try to wrap it somehow

**Pull request merged**